src/query/parser/promql/parse_test.go

```diff
@@ -459,3 +490,10 @@ func TestFailedTemporalParse(t *testing.T) {
 	_, err := Parse(q, models.NewTagOptions())
 	require.Error(t, err)
 }
+
+func TestMissingTagsDoNotPanic(t *testing.T) {
+	q := `label_join(up, "foo", ",")`
+	p, err := Parse(q, models.NewTagOptions())
+	require.NoError(t, err)
+	_, _, _ = p.DAG()
```

Comment:
nit: You could do `assert.NotPanics(t, func() { _, _, _ = p.DAG() })` to be explicit about intent of calling this method (slighty nicer output if it does actually panic too, keeps running rest of the tests in the module rather than stopping whole test run).

Reply:
Oh that's pretty cool, makes sense it's an inbuilt